### ISSUE TITLE
[auth] Add support for GameCenter and Play SignIn

### ISSUE THREAD

**Issue by Lyokone** [firebase/flutterfire#9441]
Currently, we don't support https://firebase.google.com/docs/auth/ios/game-center and https://firebase.google.com/docs/auth/android/play-games.

We need to add instruction on how to add them.

**Comment by orkun1675** [comment]
Thank you for filing this issue Lyokone, this would be very beneficial for my app.

To clarify, there is no native support for this yet, right? So it's more than a matter of adding instructions/documentation?

If there is eng work to be performed here, do you know what a rough ETA would be?

**Comment by Lyokone** [comment]
Hello,
There is currently not a native support for GameCenter. You can find the documentation for Play Sign In right here: https://github.com/firebase/flutterfire/blob/master/docs/auth/federated-auth.md#google-play-games-games

There is no ETA for the time being on Game Center

**Comment by 131e55** [comment]
I'm planning to replace Swift games with Flutter games to support Android.

Currently, is not possible to replace GameCenterAuthProvider(Swift)?

**Comment by orkun1675** [comment]
To clarify, the linked PR only adds support for siging into Firebase using credentials taken from Game Center or Play Games, right?

So we still need a 3rd party plugin that runs on iOS and Android to trigger the Game Center / Play Games login and pass back the credentials?

I'm currently using a [plugin fork](https://github.com/orkun1675/game_services_firebase_auth) I created to handle this flow but would love to deprecate that if Firebase adds internal support.

**Comment by orkun1675** [comment]
Tagging @russellwheatley since they have worked on the PR.

**Comment by russellwheatley** [comment]
Yes, you need 3rd part support for signing in with Game Center or Play games before signing in to Firebase using them as credentials.
